pkg/streamingpromql/operators/functions/factories.go

```diff
@@ -496,6 +496,39 @@ func TimestampFunctionOperatorFactory(args []types.Operator, memoryConsumptionTr
 	return operators.NewDeduplicateAndMerge(o, memoryConsumptionTracker), nil
 }
 
+func SortByLabelOperatorFactory(descending bool) InstantVectorFunctionOperatorFactory {
+	functionName := "sort_by_label"
+	if descending {
+		functionName = "sort_by_label_desc"
+	}
+
+	return func(args []types.Operator, memoryConsumptionTracker *limiter.MemoryConsumptionTracker, _ *annotations.Annotations, expressionPosition posrange.PositionRange, timeRange types.QueryTimeRange) (types.InstantVectorOperator, error) {
+		if len(args) < 2 {
+			// Should be caught by the PromQL parser, but we check here for safety.
+			return nil, fmt.Errorf("expected at least 2 arguments for %s, got %v", functionName, len(args))
+		}
+
+		inner, ok := args[0].(types.InstantVectorOperator)
+		if !ok {
+			// Should be caught by the PromQL parser, but we check here for safety.
+			return nil, fmt.Errorf("expected an instant vector for 1st argument for %s, got %T", functionName, args[0])
+		}
+
+		var labels []string
+		for i := 1; i < len(args); i++ {
+			l, ok := args[i].(types.StringOperator)
+			if !ok {
+				// Should be caught by the PromQL parser, but we check here for safety.
+				return nil, fmt.Errorf("expected a string for argument %d, got %T", i+1, args[i])
+			}
+
+			labels = append(labels, l.GetValue())
+		}
+
+		return NewSortByLabel(inner, labels, descending, memoryConsumptionTracker, expressionPosition), nil
```

Comment:
If this is a range query, should we just return `inner` here given `sort_by_label` doesn't do anything for range queries?

Reply:
Done in 6f501e635c1299a271b6d238ee0f73019c72564e